checker/src/org/checkerframework/checker/signedness/SignednessAnnotatedTypeFactory.java

```diff
@@ -154,5 +174,42 @@ public Void visitCompoundAssignment(CompoundAssignmentTree tree, AnnotatedTypeMi
             annotateBoolean(type);
             return null;
         }
+
+        @Override
+        public Void visitIdentifier(IdentifierTree tree, AnnotatedTypeMirror type) {
+            TypeMirror javaType = type.getUnderlyingType();
+            ValueAnnotatedTypeFactory valFact = getValueAnnotatedTypeFactory();
```

Comment:
For efficiency, should these two lines be executed only if javaType is a primitive?